code/modules/client/preferences2/character_save.dm

```diff
@@ -0,0 +1,475 @@
+	var/slot_locked = FALSE
+	/// Was this loaded from the DB? (This is used to decide on INSERT or UPDATE queries)
+	var/from_db = FALSE
+
+	// Character Related Vars //
+	/// Species datum
+
+	var/real_name
+	var/be_random_name = FALSE
+	var/be_random_body = FALSE
+	var/gender = MALE
+	var/age = 30
+	var/underwear = "Nude"
+	var/underwear_color = "000"
+	var/undershirt = "Nude"
+	var/socks = "Nude" // how lewd
+	var/helmet_style = HELMET_DEFAULT
+	var/backbag = DBACKPACK
+	var/jumpsuit_style = PREF_SUIT
+	var/hair_style = "Bald"
+	var/hair_color = "000"
+	var/gradient_color = "000"
+	var/gradient_style = "None"
+	var/facial_hair_style = "Shaved"
+	var/facial_hair_color = "000"
+	var/skin_tone = "caucasian1"
+	var/eye_color = "000"
+	var/datum/species/pref_species
+	var/list/features = list(
+						"body_size" = "Normal",
+						"mcolor" = "FFF",
+						"ethcolor" = "9c3030",
+						"tail_lizard" = "Smooth",
+						"tail_human" = "None",
+						"snout" = "Round",
+						"horns" = "None",
+						"ears" = "None",
+						"wings" = "None",
+						"frills" = "None",
+						"spines" = "None",
+						"body_markings" = "None",
+						"legs" = "Normal Legs",
+						"moth_wings" = "Plain",
+						"ipc_screen" = "Blue",
+						"ipc_antenna" = "None",
+						"ipc_chassis" = "Morpheus Cyberkinetics(Greyscale)",
+						"insect_type" = "Common Fly"
+					)
+	var/list/custom_names = list()
+	var/preferred_ai_core_display = "Blue"
+	var/preferred_security_department = SEC_DEPT_RANDOM
+	var/list/all_quirks = list()
+	var/list/job_preferences = list()
+	var/list/equipped_gear = list()
+	var/joblessrole = BERANDOMJOB  //defaults to 1 for fewer assistants
+	var/uplink_spawn_loc = UPLINK_PDA
+
+
+/datum/character_save/New()
+	real_name = get_default_name()
+	for(var/custom_name_id in GLOB.preferences_custom_names)
+		custom_names[custom_name_id] = get_default_name(custom_name_id)
+
+#define SAFE_READ_QUERY(idx, target)  if(Q.item[idx]) target = Q.item[idx]
+
+/datum/character_save/proc/handle_query(datum/DBQuery/Q)
+	from_db = TRUE
+
+	// please keep these in numerical order I beg
+	//Species
+	var/species_id
+	SAFE_READ_QUERY(2, species_id)
+	if(species_id)
+		var/newtype = GLOB.species_list[species_id]
+		if(newtype)
+			pref_species = new newtype
+
+	//Character
+	SAFE_READ_QUERY(3, real_name)
+	SAFE_READ_QUERY(4, be_random_name)
+	SAFE_READ_QUERY(5, be_random_body)
+	SAFE_READ_QUERY(5, gender)
+	SAFE_READ_QUERY(6, age)
+	SAFE_READ_QUERY(7, hair_color)
+	SAFE_READ_QUERY(8, gradient_color)
+	SAFE_READ_QUERY(9, facial_hair_color)
+	SAFE_READ_QUERY(10, eye_color)
+	SAFE_READ_QUERY(11, skin_tone)
+	SAFE_READ_QUERY(12, hair_style)
+	SAFE_READ_QUERY(13, gradient_style)
+	SAFE_READ_QUERY(14, facial_hair_style)
+	SAFE_READ_QUERY(15, underwear)
+	SAFE_READ_QUERY(16, underwear_color)
+	SAFE_READ_QUERY(17, undershirt)
+	SAFE_READ_QUERY(18, socks)
+	SAFE_READ_QUERY(19, backbag)
+	SAFE_READ_QUERY(20, jumpsuit_style)
+	SAFE_READ_QUERY(21, uplink_spawn_loc)
```

Comment:
I think all of these may be offset by 1, is it not possible to read by using the key of the pref?
![image](https://user-images.githubusercontent.com/26465327/191828114-d327e271-695c-4c14-94c2-7e0e6705e339.png)

Reply:
We noticed that the colours seem to be swapped, it appears that its reading 1 off somewhere at least

Reply:
@Crossedfall noticed that 5 is duplicated.
![image](https://user-images.githubusercontent.com/26465327/191828401-2130104d-91d5-4fcf-a5eb-1c707d12b0bf.png)

Reply:
Tested.. Pushing everything up a value so the final read is correctly set to `31` resolves loading issues (when the `SELECT` query is correct)